pcsx2/Recording/PadData.cpp

```diff
@@ -0,0 +1,274 @@
+#include "PrecompiledHeader.h"
+
+#include "Common.h"
+#include "ConsoleLogger.h"
+#include "PadData.h"
+
+PadData::PadData()
+{
+	for (int port = 0; port < 2; port++)
+	{
+		buf[port][0] = 255;
+		buf[port][1] = 255;
+		buf[port][2] = 127;
+		buf[port][3] = 127;
+		buf[port][4] = 127;
+		buf[port][5] = 127;
+	}
+}
+
+void PadData::logPadData(u8 port, u16 bufCount, u8 buf[512]) {
+	// skip first two bytes because they dont seem to matter
+	if (port == 0 && bufCount > 2) 
+	{
+		if (bufCount == 3)
+		{
+			controlLog(wxString::Format("\nController Port %d", port));
+			controlLog(wxString::Format("\nPressed Flags - "));
+		}
+		if (bufCount == 5) // analog sticks
+		{
+			controlLog(wxString::Format("\nAnalog Sticks - "));
+		}
+		if (bufCount == 9) // pressure sensitive bytes
+		{
+			controlLog(wxString::Format("\nPressure Bytes - "));
+		}
+		controlLog(wxString::Format("%3d ", buf[bufCount]));
+	}
+}
+
+std::vector<wxString> split(const wxString &s, char delim) {
+	std::vector<wxString> elems;
+	wxString item;
+	for (char ch : s) {
+		if (ch == delim) {
+			if (!item.empty())
+				elems.push_back(item);
+			item.clear();
+		}
+		else {
+			item += ch;
+		}
+	}
+	if (!item.empty())
+		elems.push_back(item);
```

Comment:
Would it not be more efficient to find the delimiter and then extract the substring instead of appending a char to the string one by one until the delimiter is found?